tests/test_requests.py

```diff
@@ -526,6 +526,9 @@ def test_proxy_error(self):
         with pytest.raises(ProxyError):
             requests.get('http://localhost:1', proxies={'http': 'non-resolvable-address'})
 
+        with pytest.raises(ProxyError):
```

Comment:
It may be easier to split this into its own test. Otherwise, let's at least add a comment stating why this should fail differently than the one above.

Reply:
I added 4 test cases in the new commit as a different test.